elastic/kibana
# [Security Solution][Lists] - Update esClient header

## Summary

Goal of this PR is to reduce deprecation warning noise originating from Elastic products. 

- Updates some of the deprecated esClient request params like:
  -  `esClient.index`: param `body` renamed to `document`
  - `esClient.bulk`: param `body` renamed to `operations`
  - `esClient.deleteByQuery`: param `body` no longer exists, body params removed from nested `body` structure
  - `esClient.count`: param `body` no longer exists, body params removed from nested `body` structure
  - `esClient.search`: param `body` no longer exists, body params removed from nested `body` structure
    - there were some where I wasn't too sure how the old values mapped to the new and in those cases I left those alone 
  - `esClient.update`: param `body` no longer exists, body params removed from nested `body` structure
- Added unit tests ensuring header is added to esClient requests, might seem overkill, but wanted to ensure that header wasn't accidentally removed in the future.


### Testing

- Make sure to run the latest ES snapshot (yarn es snapshot --license trial) with Kibana
- Navigate to Upgrade Assistant
- Under "Review deprecated settings and resolve issues" click on the "Elasticsearch deprecation logs" link
- On an empty cluster there shouldn't be any deprecation in the counter
- Navigate to both the security solution product and make sure that the filters are applied to not include Elastic products
- Open Console in a new tab and add a deprecation with a product that does not exists
```JSON
POST .logs-deprecation.elasticsearch-default/_doc/
{
  "event.dataset" : "deprecation.elasticsearch",
  "@timestamp" : "2022-12-25T13:51:40,618Z", // Date in the future
  "log.level" : "WARN",
  "log.logger" : "org.elasticsearch.deprecation.cluster.metadata.IndexNameExpressionResolver",
  "elasticsearch.cluster.name" : "elasticsearch",
  "elasticsearch.cluster.uuid" : "4q7OmaQkT-S0v-GJogb7yQ",
  "elasticsearch.node.id" : "qY6aIkdIQTiAOdKbqMiQCg",
  "elasticsearch.node.name" : "Elastic.Me",
  "trace.id" : "",
  "message" : "this request accesses system indices: [.foo], but in a future major version, direct access to system indices will be prevented by default",
  "data_stream.type" : "logs",
  "data_stream.dataset" : "deprecation.elasticsearch",
  "data_stream.namespace" : "default",
  "ecs.version" : "1.7",
  "elasticsearch.event.category" : "api",
  "event.code" : "open_system_index_access",
  "elasticsearch.http.request.x_opaque_id" : "foo",
  "elasticsearch.elastic_product_origin" : "other" // Give any unknown name here
}
```
- Back in Upgrade assistant, the counter should have incremented by 1
- Back in console add a new doc but this time set the "elasticsearch.elastic_product_origin" : "security"
- Back in Upgrade assistant the counter should stay the same

### Checklist


### For maintainers
